Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm

```diff
@@ -348,6 +350,15 @@ void didExceedMemoryFootprintThreshold(size_t footprint, const String& domain, u
     bool m_hasWebCryptoMasterKeySelector { false };
 };
 
+#if PLATFORM(IOS_FAMILY)
+
+@interface _WKWebsiteDataStoreBSActionHandler : NSObject <_UIApplicationBSActionHandler>
++ (_WKWebsiteDataStoreBSActionHandler *)shared;
+- (void)setWebPushActionHandler:(WKWebsiteDataStore *(^)(_WKWebPushAction *action))handler;
```

Comment:
I wonder if requiring the WKWebsiteDataStore to be provided synchronously in the return value of the callback might be an issue for some clients, but I think we can worry about that issue later if it ever crops up.